official/datasets/movielens.py

```diff
@@ -0,0 +1,308 @@
+
+GENRE_COLUMN = "genres"
+ITEM_COLUMN = "item_id"  # movies
+RATING_COLUMN = "rating"
+TIMESTAMP_COLUMN = "timestamp"
+TITLE_COLUMN = "titles"
+USER_COLUMN = "user_id"
+
+GENRES = [
+    'Action', 'Adventure', 'Animation', "Children", 'Comedy', 'Crime',
+    'Documentary', 'Drama', 'Fantasy', 'Film-Noir', 'Horror', "IMAX", 'Musical',
+    'Mystery', 'Romance', 'Sci-Fi', 'Thriller', 'War', 'Western'
+]
+N_GENRE = len(GENRES)
+
+RATING_COLUMNS = [USER_COLUMN, ITEM_COLUMN, RATING_COLUMN, TIMESTAMP_COLUMN]
+MOVIE_COLUMNS = [ITEM_COLUMN, TITLE_COLUMN, GENRE_COLUMN]
+
+# Note: Users are indexed [1, k], not [0, k-1]
+NUM_USER_IDS = {
+    ML_1M: 6040,
+    ML_20M: 138493,
+}
+
+# Note: Users are indexed [1, k], not [0, k-1]
+# Both the 1m and 20m datasets use the same movie set.
+NUM_ITEM_IDS = 3952
+
+MAX_RATING = 5
+
+
+NUM_RATINGS = {
+    ML_1M: 1000209,
+    ML_20M: 20000263
+}
+
+
+def _download_and_clean(dataset, data_dir):
+  """Download MovieLens dataset in a standard format.
+
+  This function downloads the specified MovieLens format and coerces it into a
+  standard format. The only difference between the ml-1m and ml-20m datasets
+  after this point (other than size, of course) is that the 1m dataset uses
+  whole number ratings while the 20m dataset allows half integer ratings.
+  """
+  if dataset not in DATASETS:
+    raise ValueError("dataset must be one of: {}".format(",".join(DATASETS)))
+
+  data_subdir = os.path.join(data_dir, dataset)
+
+  expected_files = ["{}.zip".format(dataset), RATINGS_FILE, MOVIES_FILE]
+
+  tf.gfile.MakeDirs(data_subdir)
+  if set(expected_files).intersection(
+      tf.gfile.ListDirectory(data_subdir)) == set(expected_files):
+    tf.logging.info("Dataset {} has already been downloaded".format(dataset))
+    return
+
+  url = "{}{}.zip".format(_DATA_URL, dataset)
+
+  temp_dir = tempfile.mkdtemp()
+  try:
+    zip_path = os.path.join(temp_dir, "{}.zip".format(dataset))
+    def _progress(count, block_size, total_size):
+      sys.stdout.write("\r>> Downloading {} {:.1f}%".format(
+          zip_path, 100.0 * count * block_size / total_size))
+      sys.stdout.flush()
+
+    zip_path, _ = urllib.request.urlretrieve(url, zip_path, _progress)
+    statinfo = os.stat(zip_path)
+    # A new line to clear the carriage return from download progress
+    # tf.logging.info is not applicable here
+    print()
+    tf.logging.info(
+        "Successfully downloaded {} {} bytes".format(
+            zip_path, statinfo.st_size))
+
+    zipfile.ZipFile(zip_path, "r").extractall(temp_dir)
+
+    if dataset == ML_1M:
+      _regularize_1m_dataset(temp_dir)
+    else:
+      _regularize_20m_dataset(temp_dir)
+
+    for fname in tf.gfile.ListDirectory(temp_dir):
+      tf.gfile.Copy(os.path.join(temp_dir, fname),
+                    os.path.join(data_subdir, fname))
+
+  finally:
+    tf.gfile.DeleteRecursively(temp_dir)
+
+
+def _write_csv(input_path, output_path, names, skip_first, separator=","):
```

Comment:
From the comment, it seems that the function mainly does transformation, eg encoding. Should we just call it _transform_csv? or _cleanup_csv?